Fix custom layout with elementor and masonry

cristian-ungureanu opened the pull request:
### Summary
Fixes post types on blog page

### Will affect visual aspect of the product
NO


### Test instructions
- Create a custom layout in Elementor and add it to the blog page ( archive type is post )
- Enable grid display and masonry option in customizer -> layouts -> blog/archive
- Check if everything is ok
- With the same settings from 2, try to add other custom layouts on the blog page in different locations

<!-- Issues that this pull request closes. -->
Closes Codeinwp/neve-pro-addon#1803.
<!-- Should look like this: `Closes #1, #2, #3.` . -->

irinelenache commented:
@cristian-ungureanu The issue is fixed 👍 

Found some custom layouts that don't work as expected. After creating [this custom layout ](https://vertis.d.pr/J9IPEK), it will look [like this](https://vertis.d.pr/Bemzfn) when hooked to **Loop entry before, Loop entry after, Middle posts loop**

cristian-ungureanu commented:
@irinelenache Thank you for testing. I fixed the loop entry before, Loop entry after hooks, and removed the Middle posts loop hook in case masonry is enabled.
Please have another look and let me know if everything works!

irinelenache commented:
@cristian-ungureanu The mentioned issue is fixed, thanks 👍 

Found something else:

- [ ] When the **blog layout** is set to **Covers** and **Masonry** is enabled, the **Loop entry before** and **Loop entry after** hooks are part of the cover https://vertis.d.pr/3cQeMe . This wasn't happening before, but i think it would be a good idea to have the possibility to set a custom layout like this, because it doesn't look bad at all 👀 

Besides of this, everything seems to work fine 🚀
